zerver/models.py

```diff
@@ -348,6 +348,7 @@ class UserProfile(AbstractBaseUser, PermissionsMixin):
     date_joined = models.DateTimeField(default=timezone.now)
     is_mirror_dummy = models.BooleanField(default=False)
     bot_owner = models.ForeignKey('self', null=True, on_delete=models.SET_NULL)
+    starred_users = models.ManyToManyField("self")
```

Comment:
This is problematic. See the description for https://docs.djangoproject.com/en/1.9/ref/models/fields/#django.db.models.ManyToManyField

Currently it's a symmetrical relationship. To see the effect, try doing this:
In one browser, login as person A.
In another browser, or another browser session (an incognito window works if you're using Chrome), login as person B.
I'd add a test for this.

If person A stars person B, person B will implicitly star person A. Person A should not be able to modify Person B's preferences like that.